File: Editorial/102_Binary_Tree_Level_Order_Traversal/TypeScript/1_Recursion.ts
function levelOrder(root: TreeNode | null): number[][] {
    let levels: number[][] = [];
    function helper(node: TreeNode, level: number) {
        if (levels.length === level) levels.push([]);
        levels[level].push(node.val);
        if (node.left !== null) helper(node.left, level + 1);
        if (node.right !== null) helper(node.right, level + 1);
    }
    if (root !== null) helper(root, 0);
    return levels;
}

File: Editorial/102_Binary_Tree_Level_Order_Traversal/Go/1_Recursion.go
func levelOrder(root *TreeNode) [][]int {
    levels := [][]int{}
    var helper func(*TreeNode, int)
    helper = func(node *TreeNode, level int) {
        if len(levels) == level {
            levels = append(levels, []int{})
        }
        levels[level] = append(levels[level], node.Val)
        if node.Left != nil {
            helper(node.Left, level+1)
        }
        if node.Right != nil {
            helper(node.Right, level+1)
        }
    }
    if root != nil {
        helper(root, 0)
    }
    return levels
}
